### PR TITLE
*: limit chunk size to MaxChunkSize

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

File: distsql/distsql.go [hunk 1]
```diff
@@ -183,17 +188,16 @@ func (r *selectResult) getSelectResp() (*tipb.SelectResponse, error) {
...
 }
 
-func (r *selectResult) readRowsData(chk *chunk.Chunk, rowsData []byte) error {
-	var err error
-	for len(rowsData) > 0 {
+func (r *selectResult) readRowsData(chk *chunk.Chunk, rowsData []byte) (remain []byte, err error) {
+	for chk.NumRows() < r.ctx.GetSessionVars().MaxChunkSize && len(rowsData) > 0 {
 		for i := 0; i < r.rowLen; i++ {
-			rowsData, err = codec.DecodeOneToChunk(rowsData, chk, i, r.fieldTypes[i], r.loc)
+			rowsData, err = codec.DecodeOneToChunk(rowsData, chk, i, r.fieldTypes[i], r.ctx.GetSessionVars().GetTimeZone())
```
Comment (coocood) [inline distsql/distsql.go]: How about assigning TimeZone and MaxChunkSize to local variables outside the for loop?

File: distsql/distsql.go [hunk 2]
```diff
@@ -262,7 +266,7 @@ func (pr *partialResult) Close() error {
 
 // SelectDAG sends a DAG request, returns SelectResult.
 // In kvReq, KeyRanges is required, Concurrency/KeepOrder/Desc/IsolationLevel/Priority are optional.
-func SelectDAG(ctx goctx.Context, client kv.Client, kvReq *kv.Request, fieldTypes []*types.FieldType, loc *time.Location) (SelectResult, error) {
+func SelectDAG(goCtx goctx.Context, ctx context.Context, client kv.Client, kvReq *kv.Request, fieldTypes []*types.FieldType) (SelectResult, error) {
```
Comment (coocood) [inline distsql/distsql.go]: The `client` can be removed too since we have `ctx`.

**Comment by coocood** [comment]
LGTM

File: distsql/distsql.go [hunk 3]
```diff
@@ -142,18 +143,22 @@ func (r *selectResult) NextRaw() ([]byte, error) {
...
-		return errors.Trace(err)
-	}
-	if selectResp == nil {
-		return nil
-	}
-	err = r.readRowsData(chk, selectResp.Chunks[r.respChkIdx].RowsData)
-	if err != nil {
-		return errors.Trace(err)
+	for chk.NumRows() < r.ctx.GetSessionVars().MaxChunkSize {
+		selectResp, err := r.getSelectResp()
```
Comment (XuHuaiyu) [inline distsql/distsql.go]: check r.respChkIdx < len(r.selectResp.Chunks) here?

**Pull request merged**